internal/dag/cache.go

```diff
@@ -354,18 +374,15 @@ func (kc *KubernetesCache) remove(obj interface{}) bool {
 		delete(kc.httpproxydelegations, m)
 		return ok
 	case *gatewayapi_v1alpha1.Gateway:
-		m := k8s.NamespacedNameOf(obj)
-		_, ok := kc.gateways[m]
-		// TODO(youngnick): Remove this once gateway-api actually have behavior
-		// other than being removed from the cache.
-		kc.WithField("experimental", "gateway-api").WithField("name", m.Name).WithField("namespace", m.Namespace).Debug("Removing Gateway")
-		delete(kc.gateways, m)
-		return ok
+		if obj.Name == kc.gateway.Name && obj.Namespace == kc.gateway.Namespace {
```

Comment:
any reason not to use the helper here?

Reply:
Which helper are you referring to?

Reply:
The helper I added, haha. Yeah I updated makes total sense. =)